crates/re_space_view_spatial/src/space_view_2d.rs

```diff
@@ -292,18 +257,15 @@ impl SpaceViewClass for SpatialSpaceView2D {
     }
 }
 
-/// Count how many occurrences of the given component are in
-/// the intersection of the subtree and the entity bucket,
-/// but stops recursion if there is a hit.
-///
-/// * So if the subtree root is in the bucket: return 1 if it has the component, 0 otherwise.
-/// * If not: recurse on children and sum the results.
+// Count the number of entities with the given component exist that aren't
+// children of other entities in the bucket.
 fn count_non_nested_entities_with_component(
+    image_dimensions: &IntMap<EntityPath, ImageDimensions>,
     entity_bucket: &IntSet<EntityPath>,
     subtree: &EntityTree,
     component_name: &ComponentName,
 ) -> usize {
-    if entity_bucket.contains(&subtree.path) {
+    if image_dimensions.contains_key(&subtree.path) {
         // bool true -> 1
         subtree.entity.components.contains_key(component_name) as usize
```

Comment:
this method got even more confusing now. Clearly this is now only about images and the component name you passed in is just to mean image?
Let's rename it to `count_non_nested_images_with_component`